proto/spacemesh/v1/types.proto

```diff
@@ -45,8 +45,8 @@ message Activation {
 // do not include mutable data such as tx state or result.
 message Transaction {
     bytes id = 1;
-    bytes principal = 2;
-    bytes template = 3;
+    AccountId principal = 2;
+    AccountId template = 3;
```

Comment:
i don't know if template should have an hrp. this seems a bit weird. @lrettig what do you think?

Reply:
It should. It's a domain-specific thing -- the same address (i.e., bytestr) can exist on many networks, but a template may only be deployed on one, and the HRP specifies the domain.

Reply:
I think we can make it optional for both `principal` and `template` since HRP is already implied by `GENESISID`, which the tx is bound to (https://github.com/spacemeshos/pm/issues/135)